src/fv/share/tdymeshcustom.c

```diff
@@ -0,0 +1,1315 @@
+            SETERRQ(PETSC_COMM_WORLD,PETSC_ERR_USER,"Face types do not match");
+          }
+
+          PetscInt nvertices = TDyGetNumVerticesForFaceType(face_type);
+          mesh_ptr->faces.num_vertices[iconn] = nvertices;
+
+        } else {
+          SETERRQ(PETSC_COMM_WORLD,PETSC_ERR_USER,"Global face not found");
+        }
+
+        mesh_ptr->faces.cell_offset[iconn] = offset;
+        mesh_ptr->faces.cell_ids[offset++] = icell;
+        mesh_ptr->faces.cell_ids[offset++] = abs(dual_id);
+        mesh_ptr->faces.id[iconn] = cell_to_face[iface][icell];
+        mesh_ptr->faces.is_internal[iconn] = PETSC_TRUE;
+        mesh_ptr->faces.is_local[iconn] = PETSC_TRUE;
+        iconn++;
+      }
+    }
+  }
+
+  PetscFunctionReturn(0);
+}
+
+/// Computes area of triangle
+///
+/// @param [in] ugrid A TDyUGrid struct
+/// @param [in] v_id1 First vertex ID of the triangel
+/// @param [in] v_id2 Second vertex ID of the triangel
+/// @param [in] v_id3 Third vertex ID of the triangel
+/// @param [out] area Area of the triange
+///
+/// @returns 0 on success, or a non-zero error code on failure
+static PetscErrorCode ComputeArea(TDyUGrid *ugrid, PetscInt v_id1, PetscInt v_id2, PetscInt v_id3, PetscReal *area) {
+
+  PetscErrorCode ierr;
+
+  PetscReal **vertices = ugrid->vertices;
+  PetscInt dim=3;
+  PetscReal point1[dim], point2[dim], point3[dim], v1[dim], v2[dim], cross_prod[dim];
+  PetscReal dot_prod;
+
+  for (PetscInt idim=0; idim<dim; idim++) {
+    point1[idim] = vertices[v_id1][idim];
+    point2[idim] = vertices[v_id2][idim];
+    point3[idim] = vertices[v_id3][idim];
+    v1[idim] = point3[idim] - point2[idim];
+    v2[idim] = point1[idim] - point2[idim];
+  }
+  ierr = TDyCrossProduct(v1,v2,cross_prod); CHKERRQ(ierr);
+  ierr = TDyDotProduct(cross_prod,cross_prod,&dot_prod); CHKERRQ(ierr);
+
+  PetscReal magnitude = PetscPowReal(dot_prod,0.5);
+  *area = 0.5*magnitude;
+
+  PetscFunctionReturn(0);
+}
+
+/// For a triangle, compute the following:
+///  - area of the triangle
+///  - projected area of a triangle along the unit vector between upwind and downwind cells
+///  - intercept of the upwind-downwind line with the trinagle 
+///
+/// @param [in] ugrid A TDyUGrid struct
+/// @param [in] v_id1 First vertex ID of the triangel
+/// @param [in] v_id2 Second vertex ID of the triangel
+/// @param [in] v_id3 Third vertex ID of the triangel
+/// @param [in] point_up Coordinate of upwind cell
+/// @param [in] point_dn Coordinate of downwind cell
+/// @param [in] n_up_dn Unit vector joining upwind and downwind cells
+/// @param [out] area Area of the triange
+/// @param [out] area_projected Projected area along unit normal
+/// @param [out] intercept Intercept between upwind-downind line to the triangle
+///
+/// @returns 0 on success, or a non-zero error code on failure
+static PetscErrorCode ComputeAreasAndIntercpet(TDyUGrid *ugrid, PetscInt v_id1, PetscInt v_id2, PetscInt v_id3, PetscReal point_up[3], 
```

Comment:
Consider `ComputeTriAreasAndIntercept` (and watch out for the `Intercpet` typo)

Reply:
Fixed as suggested.